src/classlibnative/bcltype/system.cpp

```diff
@@ -41,14 +41,35 @@ typedef BOOL (*pfnGetPhoneVersion)(LPOSVERSIONINFO lpVersionInformation);
 pfnGetPhoneVersion g_pfnGetPhoneVersion = NULL;
 #endif
 
+typedef void(WINAPI *pfnGetSystemTimeAsFileTime)(LPFILETIME lpSystemTimeAsFileTime);
+pfnGetSystemTimeAsFileTime g_pfnGetSystemTimeAsFileTime = NULL;
+Volatile<bool> g_fGetSystemTimeAsFileTimeInitialized = false;
 
 FCIMPL0(INT64, SystemNative::__GetSystemTimeAsFileTime)
```

Comment:
Another thing that should save a few cycles is to change the signature of this method to: `void GetSystemTimeAsFileTime(INT64* pTimestamp)`.

It will make the signature of the underlying windows API match exactly the signature of the FCall. It means that the compiler can compile the FCall to just `jmp [g_pfnGetSystemTimeAsFileTime]` vs. method with a frame that it is today.